client/lib/plans/personal-plan.js

```diff
@@ -54,13 +81,18 @@ export const insertSitePersonalPlan = plans => {
 		has_domain_credit,
 		product_name,
 		product_slug,
-		raw_price
-	} = personalPlan;
+		raw_price,
+		currency_code
+	} = formatPlan( plans );
 
 	if ( ! ( isEmpty( plans ) || has( plans, product_id ) ) ) {
+		const hasCurrentPlan = some( plans, matchesProperty( 'current_plan', true ) );
```

Comment:
`insertSitePersonalPlan` and `insertPersonalPlan` are low level functions. There's not even a state to query for plans when they are called. They add the personal plan to the server response on the action creators.
In fact, that check is there so `getCurrentPlan` works with a site that has the personal plan.